File: 1. Array & String/3. String to URL/Solution.java
class Solution {
    public String replaceSpaces(String str, int length) {
        char[] chars = str.toCharArray();
        int j = chars.length;
        for (int i = length - 1; i >= 0; --i) {
            if (chars[i] == ' ') {
                chars[--j] = '0';
                chars[--j] = '2';
                chars[--j] = '%';
            } else {
                chars[--j] = chars[i];
            }
        }
        return new String(chars, j, chars.length - j);
    }
}

File: 1. Array & String/3. String to URL/Solution.cpp
class Solution {
public:
    void replaceSpaces(char *str, int length) {
        int numOfSpaces = 0;
        int i = 0, j = 0;
        for ( i = 0; i < length; ++i ) {
            if (str[i] == ' ') {
                ++numOfSpaces;
            }
        }
        
        int extendedLen  = length + 2 * numOfSpaces;
        i = extendedLen - 1;
        for( j = length - 1; j >= 0; --j ) {
            if ( str[j] != ' ' ) {
            str[i--] = str[j];
            } else {
                str[i--] = '0';
                str[i--] = '2';
                str[i--] = '%';
            }
        }
    }
};